### PR TITLE
add Donate button to About dialog

### PR DESCRIPTION
**Pull request by ronso0**
image

https://mixxx.zulipchat.com/#narrow/stream/247619-UI-.26.20UX.20design/topic/add.20Donate.20button.20to.20Help.20menu.3F

- [ ] add /donate relay at mixxx.org? so we can change the donate method/link without having to change anything else
- [ ] nice icons

### CODE REVIEW

**Comment by daschuer** [comment]
Thank you for picking it up. 
The heard is already a nice icon. 
If we have alternatives, we may randomize them to draw attention if a user opens the window a second time. Just an idea. 

The other idea is: do we want to modify the button text or a lable an online source to point out donation targets or anything else?

File: src/defs_urls.h [hunk 1]
```diff
@@ -4,6 +4,9 @@
 #define MIXXX_WEBSITE_SHORT_URL "www.mixxx.org"
 #define MIXXX_SUPPORT_URL       "https://www.mixxx.org/support/"
 #define MIXXX_TRANSLATION_URL   "https://www.transifex.com/projects/p/mixxxdj/"
+#define MIXXX_DONATE_URL              \
+    "https://www.paypal.com/cgi-bin/" \
+    "webscr?cmd=_s-xclick&hosted_button_id=QSFMYWN2B3JD2"
```
Comment (Holzhaus) [inline src/defs_urls.h:9]:
```suggestion
#define MIXXX_DONATE_URL         "https://mixxx.org/donate"
```

after https://github.com/mixxxdj/website/pull/238 has been merged

Comment (esbrandt) [reply src/defs_urls.h:9]: Unless Mixxx is transformed into a legal entity, accepting ``donations`` raises more questions than answers. It is a gray area at least.

Comment (Holzhaus) [reply src/defs_urls.h:9]: Mixxx always took donations, I can even see a donation button on our website in 2008 when I use the Wayback Machine. I do agree that we should think about founding an e.V. or joining the SFC, but I don't think this should hold back this PR.

Maybe you want to share your thoughts on these Zulip threads:
- https://mixxx.zulipchat.com/#narrow/stream/109121-core-team/topic/Conservancy
- https://mixxx.zulipchat.com/#narrow/stream/109121-core-team/topic/Mixxx.20e.2EV.2E

**Comment by ronso0** [comment]
> If we have alternatives, we may randomize them to draw attention if a user opens the window a second time. Just an idea.

yeah, this might help. if someone wants look into that, fine.
alternative: pick up the rainbow from the Mixxx logo
image

just the outline:
image

For now the plain monochrome heart is fine IMO. Like the preferences icons it appears in black or white, depending on the OS theme.

**Comment by ronso0** [comment]
Ready?

File: src/dialog/dlgabout.cpp [hunk 2]
```diff
@@ -343,6 +344,16 @@ DlgAbout::DlgAbout(QWidget* parent) : QDialog(parent), Ui::DlgAboutDlg() {
                             MIXXX_WEBSITE_URL,
                             tr("Official Website")));
 
+    if (!Color::isDimColor(palette().text().color())) {
+        btnDonate->setIcon(QIcon(":/images/heart_icon_light.svg"));
+    } else {
+        btnDonate->setIcon(QIcon(":/images/heart_icon_dark.svg"));
+    }
+    btnDonate->setText(tr("Donate"));
+    connect(btnDonate, &QPushButton::clicked, this, [this] {
```
Comment (Holzhaus) [inline src/dialog/dlgabout.cpp:353]:
```suggestion
    connect(btnDonate, &QPushButton::clicked, this, [] {
```

**Review by daschuer — APPROVED** [review]
LGTM, Thank you. 

If you add the rainbow icon, I can contribute the lines showing it by a chance of 1/8 or such. 

@Holzhaus merge?

File: src/dialog/dlgabout.cpp (hunk 2 above)
Comment (Holzhaus) [reply src/dialog/dlgabout.cpp:353]: @daschuer there's a compiler warning about unused this in the lambda expression

**Comment by daschuer** [comment]
No, there is a pending comment ...

**Comment by ronso0** [comment]
> If you add the rainbow icon, I can contribute the lines showing it by a chance of 1/8 or such.

@daschuer Go ahead, the rainbow icon is in place.

**Comment by ronso0** [comment]
> > If you add the rainbow icon, I can contribute the lines showing it by a chance of 1/8 or such.
> 
> @daschuer Go ahead, the rainbow icon is in place.

or we just merge this, and you apply that in a follow-up. I don't mind

**Review by Holzhaus — APPROVED** [review]

**Comment by Holzhaus** [comment]
Or do you mean other projects (e. g. our dependencies).

By the way, should we add a tab that lists those dependencies? Would be a nice gesture to mention and add links to them I think.

**Comment by ronso0** [comment]
_all_ dependencies? group all libqt*, also include the static libs like xwax?

**Comment by Holzhaus** [comment]
Not like that. Just a list like:

- Qt
- xwax
- KeyFinder
- SoubdTouch
- Rubberband
- modplug
- Chromaprint
- FLAC
- OGG
- Vorbis
- LAME
- FFTW
- Ebur128
- PortAudio
- PortMidi
- SndFile
- TagLib
- MP4
- FFMPEG
- HSS1394
- libshout-idjc
- SQLite
- Opus
- MAD
- ID3Tag
- WavPack
- LibUSB
- HIDAPI
- lilv

+ links to their projects.

**Comment by ronso0** [comment]
Do you imagine a second tab in About? with a 2-colum list

**Comment by ronso0** [comment]
Do you imagine a second tab in About? with a 2-colum list

**Comment by daschuer** [comment]
>> The other idea is: do we want to modify the button text or a lable an online source to point out donation targets or anything else?

> To indicate we're using PayPal? Or what do you want to point out?

The idea was something like "$ 546 left to .."  but scratch that idea. Such a lookup may violate users privacy. 

We can integrate it into an update announcer later on if we feel there is a need.

**Pull request merged**